### PR TITLE
build: add pedantic spell check to build

### PR DESCRIPTION
**Pull request by zuercher**
Adds support for fixing errors via the pedantic spell checker.
Runs pedantic comment spelling checking during builds.
Adds the location where python deps are installed during
do_ci.sh (to avoid spurious format check errors).

*Risk Level*: low
*Testing*: checks comment spelling
*Docs Changes*: n/a
*Release Notes*: n/a

Signed-off-by: Stephan Zuercher <zuercher@gmail.com>

### CODE REVIEW

**Comment by zuercher** [comment]
BTW, I expect this to fail the format check because new typos have been introduced since yesterday. I'll fix them once I see the expected failure.

**Comment by zuercher** [comment]
Seems to be working. This is ready for review.

File: tools/spelling_dictionary.txt [hunk 1]
```diff
@@ -738,3 +738,5 @@ xyz
 zag
 zig
 zlib
+recvmmsg
+OS
```
Comment (venilnoronha) [inline tools/spelling_dictionary.txt]: nit: alpha?

File: tools/check_spelling_pedantic.py [hunk 2]
```diff
@@ -73,17 +80,130 @@ def debug(s):
...
+    # Read the custom dictionary
+    words = []
+    with open(self.dictionary_file, 'r') as f:
+      words = f.readlines()
+
+    # Strip comments and blank lines.
+    words = [w for w in words if len(w) > 0 and w[0] != "#"]
+
+    # Allow acronyms and abbreviations to be spelled in lowercase.
+    # (e.g. Convert "HTTP"" into "HTTP" and "http" which also matches
```
Comment (venilnoronha) [inline tools/check_spelling_pedantic.py]: s/"HTTP""/"HTTP"

**Pull request merged**